### ISSUE TITLE
updating tool in Galaxy shows invalid screen

### ISSUE THREAD

**Issue by martenson** [galaxyproject/galaxy#1865]
# could this be caused by recent changes in dependency handling when updating? @mvdbeek

![screenshot 2016-03-04 13 20 24](https://cloud.githubusercontent.com/assets/1814954/13536087/f2ecefd8-e20b-11e5-8eea-278f9017e795.png)

after clicking 'install'

![screenshot 2016-03-04 13 21 57](https://cloud.githubusercontent.com/assets/1814954/13536114/1b218644-e20c-11e5-87df-58cd9d18537b.png)

**Comment by mvdbeek** [comment]
That's possible, do you have anything in the logs about `... removing from list of repository dependencies.`  or `... removing from list of tool dependencies`?

**Comment by martenson** [comment]
@mvdbeek nope, log does not contain `removing from list of repository`

this is on Main though, I am not sure if all your changes made it to release

**Comment by martenson** [comment]
btw I just found out that the lastz package was not even present on Main...installing now

**Comment by mvdbeek** [comment]
The other thing that I noticed is that the update often fails with an error to look up dependency from the toolshed, because it assumes the dependencies need to have a higher changeset_id then the installed one.

**Comment by martenson** [comment]
@mvdbeek yeas, something like this:

```
tool_shed.galaxy_install.install_manager ERROR 2016-03-04 12:22:31,067 Error installing tool dependency lastz version 1.02.00: Unable to locate required tool shed repository named package_lastz_1_02_00 owned by devteam with revision 1d4a5e764543.
Traceback (most recent call last):
  File "/galaxy-repl/instances/main/server/lib/tool_shed/galaxy_install/install_manager.py", line 198, in install_specified_tool_dependencies
    from_tool_migration_manager=from_tool_migration_manager )
  File "/galaxy-repl/instances/main/server/lib/tool_shed/galaxy_install/install_manager.py", line 301, in install_package
    tool_dependency_db_records=None )
  File "/galaxy-repl/instances/main/server/lib/tool_shed/galaxy_install/tool_dependencies/recipe/recipe_manager.py", line 97, in process_tag_set
    tool_dependency_db_records=tool_dependency_db_records )
  File "/galaxy-repl/instances/main/server/lib/tool_shed/galaxy_install/tool_dependencies/recipe/tag_handler.py", line 519, in process_tag_set
    from_tool_migration_manager=from_tool_migration_manager )
  File "/galaxy-repl/instances/main/server/lib/tool_shed/galaxy_install/tool_dependencies/recipe/tag_handler.py", line 507, in handle_complex_repository_dependency_for_package
    raise Exception( message )
Exception: Unable to locate required tool shed repository named package_lastz_1_02_00 owned by devteam with revision 1d4a5e764543.
```

I installed the package manually and re-installed the dependency, all green now :/

**Comment by martenson** [comment]
but the changeset_id is not incremental, is it?

**Comment by mvdbeek** [comment]
it's not, but the changeset revision is. let me lookup the method ...

**Comment by mvdbeek** [comment]
@martenson do you have [this](https://github.com/galaxyproject/galaxy/blob/f974b0f740d4ed8c44ebafde74908cdea1748a1a/lib/tool_shed/galaxy_install/tool_dependencies/recipe/tag_handler.py#L505-L505) message in the logs for lastz?

**Comment by mvdbeek** [comment]
A potential underlying cause could be the assumption that updated repositories must contain updated tool_dependencies ... Which seems a bit weird to me.
This is the relevant code:
https://github.com/galaxyproject/galaxy/blob/f974b0f740d4ed8c44ebafde74908cdea1748a1a/lib/tool_shed/galaxy_install/tool_dependencies/recipe/tag_handler.py#L375

**Comment by martenson** [comment]
@mvdbeek yeah I have that log for the pastz package, see above

**Comment by mvdbeek** [comment]
I think you forget to paste it ;)

**Comment by martenson** [comment]
I see it clearly, please refresh the page, github sometimes disconnects the thread

**Comment by mvdbeek** [comment]
~~Yes, I see it! This should be easy to fix.~~
Actually, it's not that easy ... I'll see how this can be done.

**Comment by martenson** [comment]
Hopefully addressed by #1868
